src/main/java/com/codahale/metrics/WavefrontHistogram.java

```diff
@@ -0,0 +1,166 @@
+package com.codahale.metrics;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.io.OutputStream;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * WavefrontHistogram implementation for com.codahale.metrics
+ * Caveat: Cannot use the same WavefrontHistogram registry for multiple reporters as the reporter
+ * will change the state of the DeltaCounter every time the value is reported.
+ *
+ * @author Sushant Dewan (sushant@wavefront.com).
+ */
+public class WavefrontHistogram extends Histogram implements Metric {
+
+  private final com.wavefront.sdk.entities.histograms.WavefrontHistogram delegate;
+
+  public static WavefrontHistogram get(MetricRegistry registry, String metricName) {
+    return get(registry, metricName, System::currentTimeMillis);
+  }
+
+  @VisibleForTesting
+  public static synchronized WavefrontHistogram get(MetricRegistry registry,
+                                                    String metricName,
+                                                    Supplier<Long> clock) {
+    // Awkward construction trying to fit in with Dropwizard Histogram
+    TDigestReservoir reservoir = new TDigestReservoir();
+    WavefrontHistogram tDigestHistogram = new WavefrontHistogram(reservoir, clock);
+    reservoir.set(tDigestHistogram);
+    try {
+      return registry.register(metricName, tDigestHistogram);
+    } catch(IllegalArgumentException e) {
+      Histogram existing = registry.histogram(metricName);
+      if (existing instanceof WavefrontHistogram) {
+        return (WavefrontHistogram) existing;
+      } else {
+        throw new IllegalStateException("Found non-WavefrontHistogram: " + existing);
+      }
+    }
+  }
+
+  private WavefrontHistogram(TDigestReservoir reservoir, Supplier<Long> clockMillis) {
+    super(reservoir);
+    delegate = new com.wavefront.sdk.entities.histograms.WavefrontHistogram(clockMillis);
+  }
+
+  @Override
+  public void update(int value) {
+    delegate.update(value);
+  }
+
+  @Override
+  public void update(long value) {
```

Comment:
Should we provide an update(double value) method as well?

Reply:
Yes, can add that but not supported in com.codahale.metrics.Histogram
so won't be overridden